include/zephyr/net/prometheus/pm_collector.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief Prometheus collector APIs.
+ */
+
+#ifndef ZEPHYR_INCLUDE_PM_COLLECTOR_H_
+#define ZEPHYR_INCLUDE_PM_COLLECTOR_H_
+
+#include <zephyr/net/prometheus/pm_metric.h>
+
+#include <stddef.h>
+
+struct prometheus_collector_node {
+	struct prometheus_metric *metric;
+	struct pm_collector_node *next;
+};
+
+struct prometheus_collector {
+	struct prometheus_collector_node *head;
+	size_t size;
+};
```

Comment:
If these are part of the API that applications will use, then these needs to be documented so that the generated documentation contains useful information. Same comment to all structs in the header files.

Reply:
I will add detailed comments.